pkg/alibabacloud/eni/node.go

```diff
@@ -187,11 +187,14 @@ func (n *Node) CreateInterface(ctx context.Context, allocation *ipam.AllocationA
 
 // ResyncInterfacesAndIPs is called to retrieve and ENIs and IPs as known to
 // the AlibabaCloud API and return them
-func (n *Node) ResyncInterfacesAndIPs(ctx context.Context, scopedLog *logrus.Entry) (available ipamTypes.AllocationMap, remainAvailableENIsCount int, err error) {
+func (n *Node) ResyncInterfacesAndIPs(ctx context.Context, scopedLog *logrus.Entry) (available ipamTypes.AllocationMap, stats ipam.InterfaceStats, err error) {
 	limits, limitsAvailable := n.getLimits()
 	if !limitsAvailable {
-		return nil, -1, fmt.Errorf(errUnableToDetermineLimits)
+		return nil, stats, fmt.Errorf(errUnableToDetermineLimits)
 	}
+
+	stats.NodeCapacity = limits.IPv4 * (limits.Adapters - 1)
```

Comment:
Could you leave a comment why we're subtracting one adapter? I'm not familiar with AlibabaCloud, but I assume it's because on AlibabaCloud we're not using the primary adapter for pod IPs (where as on AWS ENI, it can be used depending on the value of first-interface-index)?

Reply:
This might be incorrect. I [misunderstood](https://github.com/cilium/cilium/blob/7339b89e0db239facfc4a4777485f3c68ec414d0/pkg/alibabacloud/eni/node.go#L260-L262) to mean that we don't allocate to the primary NIC, but this just deals with creating additional NIC/Addr.

I think it should just be `stats.NodeCapacity = limits.IPv4 * limits.Adapters`

Reply:
Yeah, when doing IP allocation prep, the primary is [not considered](https://github.com/cilium/cilium/blob/7339b89e0db239facfc4a4777485f3c68ec414d0/pkg/alibabacloud/eni/node.go#L260-L262) so it gets excluded from the count.

I'll update the comment.